### PR TITLE
[vcpkg_configure_make] Respect host triplet when determining --build

### PR DESCRIPTION
**Pull request by Osyotr**
This PR fixes installation of x86 versions of make-based ports on x64 windows with `--host-triplet=x86-windows`.
This command failed previously:
`vcpkg install icu --triplet=x86-windows --host-triplet=x86-windows`

### CODE REVIEW

File: scripts/cmake/vcpkg_configure_make.cmake [hunk 1]
```diff
@@ -1,19 +1,3 @@
...
-    if(host_arch MATCHES "(amd|AMD)64")
-        set(${out_var} mingw64)
-    elseif(host_arch MATCHES "(x|X)86")
-        set(${out_var} mingw32)
-    else()
-        message(FATAL_ERROR "Unsupported mingw architecture ${host_arch} in z_vcpkg_determine_autotools_host_cpu!" )
-    endif()
-    unset(host_arch)
-endmacro()
-
```
Comment (Cheney-W) [inline scripts/cmake/vcpkg_configure_make.cmake:16]: Why did you delete this section?

Comment (Neumann-A) [reply scripts/cmake/vcpkg_configure_make.cmake:16]: It is unused.

File: scripts/cmake/vcpkg_configure_make.cmake [hunk 2]
```diff
@@ -220,8 +204,16 @@ function(vcpkg_configure_make)
...
             # Only for ports using autotools so we can assume that they follow the common conventions for build/target/host
             if(CMAKE_HOST_WIN32)
-                set(arg_BUILD_TRIPLET "--build=${BUILD_ARCH}-pc-mingw32")  # This is required since we are running in a msys
-                                                                            # shell which will be otherwise identified as ${BUILD_ARCH}-pc-msys
+                # Respect host triplet when determining --build
+                if(NOT VCPKG_CROSSCOMPILING)
+                    set(_win32_build_arch "${TARGET_ARCH}")
+                else()
+                    set(_win32_build_arch "${BUILD_ARCH}")
+                endif()
```
Comment (Neumann-A) [inline scripts/cmake/vcpkg_configure_make.cmake:212]: maybe this should be in `z_vcpkg_determine_autotools_host_cpu` ?

Comment (Osyotr) [reply scripts/cmake/vcpkg_configure_make.cmake:212]: This code requires more cleanup.
I'd like to do it in a separate PR.
